sotodlib/io/load_smurf.py

```diff
@@ -1683,6 +1703,42 @@ def get_channel_info(status, mask=None, archive=None, obsfiledb=None,
     
     return ch_info
 
+def _get_sample_info(filenames):
+    """Scan through a list of files and count samples.
+    
+    Args
+    -----
+    filenames : list
+        list of filenames
+    
+    Returns 
+    --------
+    out : list
+        a list of dictionaries formatted for load_file. 
+        in the pattern of [ {filename: filename, sample_range: (sample_start, sample_stop)}]
+    """
+    out = []
+    start = 0
+    for file in filenames:
+        samps = 0 
+        reader = so3g.G3IndexedReader(file)
+        while True:
+            frames = reader.Process(None)
+            if len(frames) == 0:
+                break
+            frame = frames[0]
+            if str(frame.type) != 'Scan':
+                continue
+            data = frame.get('data')
+            sostream_version = frame.get('sostream_version', 0)
+            if sostream_version >= 2:
+                samps += len(data.times)
+            else:
+                samps += len(data)
+        out.append( {'filename':file, 'sample_range':(start, start+samps)} )
```

Comment:
I'm pretty sure this is only ever used inside load_file, which seems to adjust the start sample to correctly match up with the sample counting inside a long g3 stream (if the file is x_019.g3, for example), but it does seem like if a file like that is passed to this function then start will always be 0. That's probably not a problem, but that should be made clear in the docstring here, so that no one gets confused about the sample counting in case they end up using this function by itself in any development.

Reply:
I'll add more to the docstrings.